File: 0845-longest-mountain-in-array/0845-longest-mountain-in-array.java
class Solution {
    public int longestMountain(int[] arr) {
        int count=0; 
        int inc=0, dec=0;
        int maxInc=0, maxDec=0;
        int i=0;
        
        while (i < arr.length - 1) {
    // 1. Check this FIRST: If we were going down, but now we are going up, 
    // it means a new mountain is starting. Reset counters first.
    if (arr[i] < arr[i + 1] && dec > 0) {
        inc = 1;
        dec = 0;
        i++;
    }
    // 2. Otherwise, if it's just a normal upward slope step
    else if (arr[i] < arr[i + 1]) {
        inc++;
        i++;
    }
    // 3. Normal downward slope step
    else if (arr[i] > arr[i + 1] && inc > 0) {
        dec++;
        i++;
        // Track the valid mountain length dynamically right here!
        count = Math.max(count, inc + dec + 1); 
    }
    // 4. Flat ground (arr[i] == arr[i+1]) or going down without a peak
    else {
        inc = 0;
        dec = 0;
        i++;
    }
}
return count;
        
    }
    
}

File: 0845-longest-mountain-in-array/0845-longest-mountain-in-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var longestMountain = function(arr) {
    let count; let maxCount=0;
        for(let i=1; i<arr.length-1; i++){

            if(arr[i-1]<arr[i] && arr[i]>arr[i+1]){
               
                count=1;
                let l=i;
                while(l>0 && arr[l]>arr[l-1]){
                    count++;
                    l--;
                }
                let r=i;
                while(r<arr.length-1 && arr[r]>arr[r+1]){
                    count++;
                    r++;
                }
                maxCount=Math.max(maxCount,count);
               
            }
          
            
        }
        return maxCount;
};